Tasks/VsTest/inputparser.ts

```diff
@@ -0,0 +1,481 @@
+        const rerunFailedTestCasesMaxLimit = parseInt(tl.getInput('rerunFailedTestCasesMaxLimit'));
+        if (!isNaN(rerunFailedTestCasesMaxLimit)) {
+            inputDataContract.ExecutionSettings.RerunSettings.RerunFailedTestCasesMaxLimit = rerunFailedTestCasesMaxLimit;
+            console.log(tl.loc('rerunFailedTestCasesMaxLimit', inputDataContract.ExecutionSettings.RerunSettings.RerunFailedTestCasesMaxLimit));
+        } else {
+            tl.warning(tl.loc('invalidRerunFailedTestCasesMaxLimit'));
+        }
+    } else {
+        const rerunFailedThreshold = parseInt(tl.getInput('rerunFailedThreshold'));
+        if (!isNaN(rerunFailedThreshold)) {
+            inputDataContract.ExecutionSettings.RerunSettings.RerunFailedThreshold = rerunFailedThreshold;
+            console.log(tl.loc('rerunFailedThreshold', inputDataContract.ExecutionSettings.RerunSettings.RerunFailedThreshold));
+        } else {
+            tl.warning(tl.loc('invalidRerunFailedThreshold'));
+        }
+    }
+
+    const rerunMaxAttempts = parseInt(tl.getInput('rerunMaxAttempts'));
+    if (!isNaN(rerunMaxAttempts)) {
+        inputDataContract.ExecutionSettings.RerunSettings.RerunMaxAttempts = rerunMaxAttempts;
+        console.log(tl.loc('rerunMaxAttempts', inputDataContract.ExecutionSettings.RerunSettings.RerunMaxAttempts));
+    } else {
+        tl.warning(tl.loc('invalidRerunMaxAttempts'));
+    }
+    return inputDataContract;
+}
+
+function getRunIdentifier(): string {
+    let runIdentifier: string = '';
+    const taskInstanceId = taskinputparser.getDtaInstanceId();
+    const dontDistribute = tl.getBoolInput('dontDistribute');
+    const releaseId = tl.getVariable('Release.ReleaseId');
+    const jobId = tl.getVariable('System.JobPositionInPhase');
+    const parallelExecution = tl.getVariable('System.ParallelExecutionType');
+    const phaseId = utils.Helper.isNullEmptyOrUndefined(releaseId) ?
+        tl.getVariable('System.PhaseId') : tl.getVariable('Release.DeployPhaseId');
+    if ((!utils.Helper.isNullEmptyOrUndefined(parallelExecution) && parallelExecution.toLowerCase() === 'multiconfiguration')
+        || dontDistribute) {
+        runIdentifier = `${phaseId}/${jobId}/${taskInstanceId}`;
+    } else {
+        runIdentifier = `${phaseId}/${taskInstanceId}`;
+    }
+
+    return runIdentifier;
+}
+
+// hydra: rename function and maybe refactor and add logic inline
+function getTIALevel(fileLevel: string) {
+    if (fileLevel && fileLevel.toUpperCase() === 'FALSE') {
+        return false;
+    }
+    return true;
+}
+
+function getTestPlatformPath(inputDataContract : idc.InputDataContract) {
+    let vsTestVersion = tl.getInput('vsTestVersion');
+    if (vsTestVersion.toLowerCase() === 'latest') {
+        // latest
+        tl.debug('Searching for latest Visual Studio');
+        const vstestconsole15Path = versionfinder.getVSTestConsole15Path();
+        if (vstestconsole15Path) {
+            vsTestVersion = '15.0';
```

Comment:
remove line

Reply:
check for manual vstest.console.exe path where cx can give full path or dir

Reply:
done in the invoking function above